storage-api/src/java/org/apache/hadoop/hive/common/io/DiskRangeList.java

```diff
@@ -228,6 +228,16 @@ public long getTotalLength() {
     return result;
   }
 
+  @Override
+  public int hashCode() {
+    return super.hashCode();
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    return super.equals(other);
```

Comment:
this equals doesn't care about next/prev fields

Reply:
Addressed as part of 0e40c954e3aa7af7e932b1b26091012707b80fe2
As discussed, let's make sure we update such methods across packages.